dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -225,6 +225,67 @@ def new_post
       render :new_general
     end
 
+    # GET /pd/workshop_survey/csf/:survey_name/:enrollment_code
+    def new_csf
+      params[:survey_name] = params[:survey_name]&.downcase
+      return render_404 unless CSF_SURVEY_INDEXES.key? params[:survey_name]
+
+      if params[:survey_name] == PRE_DEEPDIVE_SURVEY
+        # TODO: what if we don't require log-in? what if pre-ws survey can be annonymous
+        workshop = Workshop.
+          where(course: COURSE_CSF).
+          where(subject: SUBJECT_CSF_201).
+          enrolled_in_by(current_user).nearest
```

Comment:
Just a style question: Do you prefer the separate `where` calls to multiple arguments?
```rb
workshop = Workshop.
  where(course: COURSE_CSF, subject: SUBJECT_CSF_201).
  enrolled_in_by(current_user).nearest
```

Reply:
Ah I don't know that multiple conditions in `where` is possible. Will update.